.github/actions/bump-version/action.yml

```diff
@@ -0,0 +1,149 @@
+  type:
+    description: Type of bump. Equivalent to the input `bump-type`.
+    value: ${{ steps.results.outputs.type }}
+runs:
+  using: composite
+  steps:
+    - name: Check Versioning Scheme Validity
+      shell: bash
+      env:
+        VALID_VERSIONING_SCHEMES: calver-minor semver-major-minor
+
+      run: |
+        # check validity of versioning schemes
+        found=false
+        for valid_scheme in ${{ env.VALID_VERSIONING_SCHEMES }}; do
+          if [[ "${{ inputs.versioning-scheme }}" == "${valid_scheme}" ]]; then
+            found=true
+            break
+          fi
+        done
+        if [[ "${found}" == "false" ]]; then
+          echo "::error::${{ inputs.versioning-scheme }} is not supported by this action. Please check the list of allowed 'versioning-scheme's."
+          exit 1
+        fi
+
+    - name: Check Bump Type Validity for ${{ inputs.versioning.scheme }}
+      shell: bash
+      env:
+        VALID_CALVER_MINOR_BUMP_TYPES: major minor
+        VALID_SEMVER_MAJOR_MINOR_BUMP_TYPES: major minor
+      run: |
+        # check validity of bump types for each versioning scheme
+        found=false
+
+        if [[ "${{ inputs.versioning-scheme }}" == "calver-minor" ]]; then
+          for valid_bump in ${{ env.VALID_CALVER_MINOR_BUMP_TYPES }}; do
+            if [[ "${{ inputs.bump-type }}" == "${valid_bump}" ]]; then
+              exit 0
+            fi
+          done
+        elif [[ "${{ inputs.versioning-scheme }}" == "semver-major-minor" ]]; then
+          for valid_bump in ${{ env.VALID_SEMVER_MAJOR_MINOR_BUMP_TYPES }}; do
+            if [[ "${{ inputs.bump-type }}" == "${valid_bump}" ]]; then
+              exit 0
+            fi
+          done
+        fi
+
+        echo "::error::Bump type ${{ inputs.bump-type }} is not supported in ${{ inputs.versioning-scheme }}. Please check the list of allowed 'bump-type's."
+        exit 1
+
+    - name: Bump - calver-minor
+      shell: bash
+      if: inputs.versioning-scheme == 'calver-minor'
+      id: calver-minor
+      # Regarding the regex in the script: `([0-9]{4}\.(0[1-9]|1[1-2]))\.([0-9]+)` is broken down into:
+      # `([0-9]{4}\.(0[1-9]|1[1-2]))`: Major version as a date in the form YYYY.0M (eg. `2024.03`)
+      # `\.`: Version separator (eg. `.`)
+      # `([0-9]+)`: Minor version (eg. `1`)
+      # which would give `2024.03.1`
+      run: |
+        regex="([0-9]{4}\.(0[1-9]|1[1-2]))\.([0-9]+)"
+        if [[ "${{ inputs.version }}" =~ "${regex}" ]]; then
+          major_version="${BASH_REMATCH[1]}"
+          minor_version="${BASH_REMATCH[3]}"
+        else
+          echo "::error::${{ inputs.version }} does not conform to ${{ inputs.versioning-scheme }}!"
+          exit 1
+        fi
+
+        if [[ "${{ inputs.bump-type }}" == "minor" ]]; then
+          minor_version=$((minor_version + 1))
+        elif [[ "${{ inputs.bump-type }}" == "major" ]]; then
+          major_version=$(date +%Y.%m)
+          minor_version=0
```

Comment:
The responsibility for checking if a `major_version` already exists falls to the workflow that is calling this one?

Reply:
Yes, it should. 
But this has unearthed another aspect that we need...we don't have a mechanism to just do (latest tag month + 1). We could add a new type of bump for `calver-minor` called `specific` that takes in a date and does `YYYY.(0M+1).0`.

Reply:
The Ci needs to check if a `calver` version exists already, so it knows it needs to bump the date. Are we expecting the dev to supply the date or will `specific` work it out for itself?

Reply:
Added this functionality (except `major` now does month+1, and `current` does the current day)